modules/client-handler/src/test/java/org/apache/ignite/client/handler/ClientHandlerIntegrationTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.client.handler;
+
+import io.netty.channel.ChannelFuture;
+import org.apache.ignite.app.Ignite;
+import org.apache.ignite.configuration.annotation.ConfigurationType;
+import org.apache.ignite.configuration.schemas.clientconnector.ClientConnectorConfiguration;
+import org.apache.ignite.internal.configuration.ConfigurationRegistry;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.msgpack.core.MessagePack;
+import org.slf4j.helpers.NOPLogger;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.util.Collections;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Client connector integration tests with real sockets.
+ */
+public class ClientHandlerIntegrationTest {
+    /** Magic bytes. */
+    private static final byte[] MAGIC = new byte[]{0x49, 0x47, 0x4E, 0x49};
+
+    private ChannelFuture serverFuture;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        serverFuture = startServer();
+    }
+
+    @AfterEach
+    public void tearDown() throws Exception {
+        serverFuture.cancel(true);
+        serverFuture.await();
+    }
+
+    @Test
+    void testHandshakeInvalidMagicHeaderDropsConnection() throws Exception {
+        try (var sock = new Socket("127.0.0.1", 10800)) {
+            OutputStream out = sock.getOutputStream();
+            out.write(new byte[]{63, 64, 65, 66, 67});
+            out.flush();
+
+            assertThrows(IOException.class, () -> writeAndFlushLoop(sock));
+        }
+    }
+
+    @Test
```

Comment:
Test description javadoc.

Reply:
* We have javadoc check suite, it does not require javadoc on tests
* Many tests don't have javadoc
* This test has a descriptive name, javadoc will only add noise